src/JoinAlgo/NestedLoopJoin.cpp

```diff
@@ -0,0 +1,32 @@
+//
+// Created by tony on 19/03/22.
+//
+
+#include <JoinAlgo/NestedLoopJoin.h>
+using namespace INTELLI;
+size_t NestedLoopJoin::join(TuplePtr *ts, TuplePtr tr, size_t tsLen, int threads) {
+  assert(threads > 0);
+  size_t result = 0;
+  for (size_t i = 0; i < tsLen; i++) {
+    if (ts[i]->key == tr->key) {
+      result++;
+    }
+  }
+  // cout<<result<<endl;
+  return result;
+}
+size_t NestedLoopJoin::join(TuplePtr *ts, TuplePtr *tr, size_t tsLen, size_t trLen, int threads) {
```

Comment:
Shared Pointer with Pointer?

Reply:
Use it as an array, legacy way for the ease of partition,etc.

Reply:
Don't do this. Use the vector class to represent array to avoid explicit pointer.

Reply:
We can not simply replace this with vector. Talk about it this afternoon..